pom.xml

```diff
@@ -0,0 +1,194 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <packaging>pom</packaging>
+  <!-- KNOWN ISSUES:
+	"Production" jar/tar files?
+	Doclet and taglet (taglet working, but only for ./xalan?)
+	stylebook (currenly, see stylebook.docgen.sh)
+       
+       xalan-test compatibility/success/submodule
+
+       NOTE: To get dependency tree from a multi-module project, use the 
+       command "mvn compile dependency:tree" so everything is in scope.
+  -->
+  
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>apache-xalan-j-2.7.3-maint-main</groupId>
+  <artifactId>apache-xalan-j-2.7.3-maint-main</artifactId>
```

Comment:
This is probably not what you want for your parent coordinates. Considering your existing artifacts at [xalan:xalan](https://search.maven.org/artifact/xalan/xalan/2.7.3/jar), your groupId should probably stay `xalan`, for everything, and neither should embed a version, it has a separate version field. I would suggest artifactId be `xalan-parent` here.

Reply:
Thanks for the suggestion. I wasn't sure how to handle that, given that what I've made sub-modules was just subdirectories broken out into separate jar files by the ant build script.

Not sure how best to handle the version number so it gets into all the right file names and fields. It looks like Maven generates a Version class for the latter, which we could wrap our existing Version interface around...?

Reply:
> It looks like Maven generates a Version class for the latter, which we could wrap our existing Version interface around...?

I'm not sure what this means. Is the existing Xalan build generating some Version interface to make the artifact version discoverable programmatically?

I can tell you that Maven artifacts are versioned, and a parent pom is versioned like an independent artifact. (One could build and publish a parent pom in a separate project, for instance, and then refer to it from many projects.) But, if you set 2.7.3 as the version on the parent pom, and don't specify different artifact versions in the module poms, they will all use the parent pom version at build time. Multi-module builds can be complex and tricky, so some reading is recommended: https://maven.apache.org/guides/mini/guide-multiple-modules.html

Reply:
I think I'd favor xalan-project over xalan-parent. The intent really is that these really are owned and maintained by a single Apache project/team.

I'm also wondering whether the group ID should really be xalan as it has been, versus apache-xalan or even apache. I don't know what the convention has been for other Apache projects, but as I understand the Maven semantics group ID should, ideally, reflect the group responsible for the package's development and maintenance. Yes,  Maven Central has a field for that, and the Java package names also communicate that, but this still seems the right way to organize the Maven packaging namespace. (xalan:xalan seems redundant, which helps push me in this direction.) Am I misunderstanding Maven intent, or Maven practice?

Reply:
> I think I'd favor xalan-project over xalan-parent.

Okay. `*-parent` is just frequently used, and I'd say more idiomatic for parent poms. For example: https://github.com/apache/maven-resolver/blob/master/pom.xml#L25

> I'm also wondering whether the group ID should really be xalan as it has been, versus org.apache.xalan or even org.apache.

Since xalan already has artifacts published at Maven Central, I'd suggest maintaining the existing coordinates. If you do change the groupId, it would be more conventional to follow the reverse domain pattern: https://central.sonatype.org/publish/requirements/coordinates/

But xalan has essentially been grandfathered in with an old groupId, from when folks were just using their project name as a group, e.g. [junit:junit](https://search.maven.org/artifact/junit/junit/4.13.2/jar). If you change it now, you'll need to do more work to make sure Maven knows both sets of coordinates are the same artifact so dependency resolution will work for downstream users.

Reply:
OK. Fixed the IDs. 

Also fixed some (x)html which was using uppercase tagnames, which HTML may have treated as identical to lowercase but XHTML doesn't.

Thanks for the catch. Any other errors, functional or stylistic?